pyrobosim/pyrobosim/gui/world_canvas.py

```diff
@@ -402,9 +402,9 @@ def navigate(self, robot, goal):
 
         # Find a path, or use an existing one, and start the navigation thread.
         if robot and robot.path_planner:
-            robot.current_goal = goal
-            goal_pose = self.world.graph_node_from_entity(goal, robot=robot).pose
-            path = robot.plan_path(robot.pose, goal_pose)
+            goal_node = self.world.graph_node_from_entity(goal, robot=robot)
+            robot.current_goal = goal_node.parent
+            path = robot.plan_path(robot.pose, goal_node.pose)
```

Comment:
I know this was here before, but I wonder if it's more principled to use [`resolve_to_location()`](https://github.com/sea-bass/pyrobosim/blob/8bd74e694d88223c62fce5d2b50349332074aa64/pyrobosim/pyrobosim/utils/knowledge.py#L170) here instead.

It may prevent you from having to get a graph node just to unpack it.

Reply:
The `resolve_to_location()` methods returns

```
    :return: The location or object spawn that meets the category and/or room filters, or None.
    :rtype: :class:`pyrobosim.core.locations.Location`/:class:`pyrobosim.core.locations.ObjectSpawn`
```

But goal is already an `Entity` since it is converted in `PyRoboSimMainWindow.on_navigate_click()`
The reason I used resolve to `graph_node_from_entity()` was because that is how it is done in the [`World.find_path()`](https://github.com/sea-bass/pyrobosim/blob/d4340278d21a003bcaadedda1f680bcaf96d7123/pyrobosim/pyrobosim/core/world.py#L760) in the  [ `WorldCanvas.navigate()` ](https://github.com/sea-bass/pyrobosim/blob/d4340278d21a003bcaadedda1f680bcaf96d7123/pyrobosim/pyrobosim/gui/world_canvas.py#L422) in `main`